safehttp/plugins/xsrf/angular/xsrf.go

```diff
@@ -0,0 +1,129 @@
+	TokenHeaderName string
+}
+
+var _ safehttp.Interceptor = &Interceptor{}
+
+// Default creates an Interceptor with TokenCookieName set to XSRF-TOKEN and
+// TokenHeaderName set to X-XSRF-TOKEN, their default values. However, in order
+// to prevent collisions when multiple applications share the same domain or
+// subdomain, each application should set a unique name for the cookie.
+//
+// See https://docs.angularjs.org/api/ng/service/$http#cross-site-request-forgery-xsrf-protection for more details.
+func Default() *Interceptor {
+	return &Interceptor{
+		TokenCookieName: "XSRF-TOKEN",
+		TokenHeaderName: "X-XSRF-TOKEN",
+	}
+}
+
+// Before checks for the presence of a matching XSRF token, generated on the
+// first page access, in both a cookie and a header. Their names should be set
+// when the Interceptor is created.
+func (it *Interceptor) Before(w *safehttp.ResponseWriter, r *safehttp.IncomingRequest, _ safehttp.InterceptorConfig) safehttp.Result {
+	if xsrf.StatePreserving(r) {
+		return safehttp.NotWritten()
+	}
+
+	c, err := r.Cookie(it.TokenCookieName)
+	if err != nil || c.Value() == "" {
+		return w.WriteError(safehttp.StatusForbidden)
+	}
+
+	tok := r.Header.Get(it.TokenHeaderName)
+	if tok == "" || tok != c.Value() {
+		// Only JavaScript running on the user domain can read the
+		// cookie and correctly set the token header. Hence, if the same token
+		// is found in both the cookie and the header, this guarantees the
+		// request came from the user's domain.
+		return w.WriteError(safehttp.StatusUnauthorized)
+	}
+
+	return safehttp.NotWritten()
+}
+
+func (it *Interceptor) addTokenCookie(w *safehttp.ResponseWriter) error {
+	tok := make([]byte, 20)
+	if _, err := rand.Read(tok); err != nil {
+		return fmt.Errorf("crypto/rand.Read: %v", err)
+	}
+	c := safehttp.NewCookie(it.TokenCookieName, base64.StdEncoding.EncodeToString(tok))
+
+	c.SetSameSite(safehttp.SameSiteStrictMode)
+	c.SetPath("/")
+	d := 24 * time.Hour
+	timeout, err := strconv.Atoi(fmt.Sprintf("%.0f", d.Seconds()))
+	if err != nil {
+		return err
+	}
+	// Set the duration of the token cookie to 24 hours.
+	c.SetMaxAge(timeout)
+	// Needed in order to make the cookie accessible by JavaScript
+	// running on the user's domain.
+	c.DisableHTTPOnly()
+
+	if err := w.SetCookie(c); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Commit generates a cryptographically secure random cookie on the first state
+// preserving request (GET, HEAD or OPTION) and sets it in the response. On
+// every subsequent request the cookie is expected alongside a header that
+// matches its value.
+func (it *Interceptor) Commit(w *safehttp.ResponseWriter, r *safehttp.IncomingRequest, resp safehttp.Response, _ safehttp.InterceptorConfig) safehttp.Result {
+	if c, err := r.Cookie(it.TokenCookieName); err == nil && c.Value() != "" {
+		return safehttp.NotWritten()
+	}
+
+	if !xsrf.StatePreserving(r) {
+		return w.WriteError(safehttp.StatusForbidden)
```

Comment:
```suggestion
		// This should never happen, as if this is a state-changing request and lacks the cookie
		// it would've been already rejected by Before.
		return w.WriteError(safehttp.StatusInternalServerError)
```